compiler/src/iree/compiler/Codegen/Utils/LinkingUtils.cpp

```diff
@@ -312,6 +312,10 @@ LogicalResult linkExecutablesInto(
       }
 
       // Merge the existing module into the new linked module op.
+      if (!variantOp.getInnerModule() || !linkedTargetOp.getInnerModule()) {
+        continue;
+      }
+
```

Comment:
you want to do this earlier on - by here we've already modified some things
the variantOp check can go right at the top of the `for (auto variantOp : variantOps) {` loop
linkedTargetOp could go at the very top of the function